engine/runtime/src/main/java/org/enso/interpreter/runtime/data/hash/HashMapToVectorNode.java

```diff
@@ -71,7 +74,9 @@ private static Object createEntriesVectorFromForeignMap(
           HashEntriesVector.createFromKeysAndValues(keys, values)
       );
     } catch (UnsupportedMessageException | StopIterationException | InvalidArrayIndexException e) {
-      throw new IllegalStateException("hashMap: " + hashMap + " has probably wrong hash interop API", e);
+      CompilerDirectives.transferToInterpreter();
+      var msg = "hashMap: " + hashMap + " has probably wrong hash interop API";
+      throw new PanicException(Text.create(msg), mapInterop);
```

Comment:
I think we shouldn't be throwing raw `Text` instances as panics. It is allowed, but quite annoying for users. Could we wrap this into some Enso type? e.g. `Assertion_Error`?

---

I did not get a chance to get to this PR before it was merged, so commenting afterwards.

Reply:
Done in https://github.com/enso-org/enso/pull/8448#discussion_r1416627776